### DISCUSSION TITLE
Delayed notifications for high glucose alerts

### DISCUSSION THREAD

**Discussion by lfagotbouquet** [NightscoutFoundation/xDrip#2133] [Q&A]
Hi,

I'm using xdrip+ on an Android 10 smartphone (last stable build, version 33be05a-2022.01.31).

I'm experiencing issues during the night, as my notifications for high glucose alerts appear delayed (no vibration or sound, except when the screen is turn on several hours later).
Also this happens only for high glucose alerts, not low glucose alerts...

I tried to modify a lot of system settings during the past weeks without success (remove wellbeing mode, disable do not disturb mode, check notifications autorisations, check any battery optimisation for xdrip, reinstall completely xdrip from the last stable version...).
Do you have some ideas or workaround to fix this?

In my opinion I'm not sure it's directly a bug from xdrip+, more some setting on my device that is muting temporarily the high alerts notifications when the screen is off for a long time.
What is very surprising to me is the fact that the low glucose alerts are working fine, maybe they have a more important priority?

Let me know if you have any suggestion to help me with this issue, it's becoming pretty annoying for me as I'm frequently having very high glucose levels during the night...

Loïc

**Comment by Navid200** [comment]
Can you bring up the settings of one of your high alerts that don't work and show a screenshot?

Go to the list of alerts at:
Settings &#8722;> Alarms and Alerts &#8722;> Glucose Level Alerts List

Long press the alert.

**Reply by lfagotbouquet** [reply]
Below are the screenshots for one of the high alert

![Screenshot_20220519-090437](https://user-images.githubusercontent.com/54840765/169231831-58c6985c-a8dc-4b10-b502-7f346b8e0a3d.png)
![Screenshot_20220519-090425](https://user-images.githubusercontent.com/54840765/169231836-aa2e8641-31c9-4019-8c68-493ab3c37c35.png)

**Reply by Navid200** [reply]
You've set the snooze to 2 hours.  Is that intentional?
Are you sure when you expected to hear the alert and didn't was not within that period?
Do you have the same for your low alerts that you say behave differently?

**Reply by lfagotbouquet** [reply]
Here are some screenshots for a low alert.
The 120 minutes snoozing was indeed intentional.

I think the only differences between the high and low alert configurations are the threshold, the sound and the snoozing time.
![Screenshot_20220520-063217](https://user-images.githubusercontent.com/54840765/169451393-ff75a438-816a-42c6-bf3d-84126d8463a3.png)
![Screenshot_20220520-063223](https://user-images.githubusercontent.com/54840765/169451396-483b7224-57c9-4e56-a0f7-17e7f3e94f5f.png)

**Comment by Navid200** [comment]
Please also check the following.
Go to Android (not xDrip) Settings &#8722;> Apps & notifications &#8722;> Notifications &#8722;> xDrip+ 
What do you see?

**Reply by Navid200** [reply]
Please show what I have asked for here.  What you have described could easily happen if you have enabled notification channels.  
Please show what I have explained in Android settings.

**Reply by lfagotbouquet** [reply]
For the notification settings for xdrip I have this
![Screenshot_20220520-063837](https://user-images.githubusercontent.com/54840765/169451816-08b5bd2e-09a0-41d2-b71a-22825ebdb220.png)
![Screenshot_20220520-063900](https://user-images.githubusercontent.com/54840765/169451822-1136ecec-91a4-46e2-b9ac-eca64c384c3a.png)
![Screenshot_20220520-063916](https://user-images.githubusercontent.com/54840765/169451823-9b446e19-78b9-4b41-900e-52a28f39fa6f.png)
![Screenshot_20220520-063935](https://user-images.githubusercontent.com/54840765/169451824-4110d1e7-5a75-4181-811a-e8c38e8e0748.png)

**Reply by Navid200** [reply]
It looks like you have enabled notification channels.  Is that intentional?
You can see it at the following and scrolling down:

Settings &#8722;> Alarms and Alerts &#8722;> Glucose Alerts Settings

**Reply by lfagotbouquet** [reply]
I tried to enable the notification channel in the xdrip settings a few days ago to check if it was working better with this option
I can try again without it next night.

Same screenshots in English
![Screenshot_20220520-064239](https://user-images.githubusercontent.com/54840765/169452576-b6750921-e3c1-4988-a425-f6015c1b0af8.png)
![Screenshot_20220520-064250](https://user-images.githubusercontent.com/54840765/169452584-bd7c6526-b3bb-412a-bb4e-6047901e4f91.png)
![Screenshot_20220520-064304](https://user-images.githubusercontent.com/54840765/169452586-be2a946c-da00-4018-adc4-7b78167e8ea0.png)
![Screenshot_20220520-064313](https://user-images.githubusercontent.com/54840765/169452587-6cbc0aa9-b56f-41ae-9e22-504699e20f7e.png)

**Comment by Navid200** [comment]
@tzachi-dar Is there any reason a high alert would behave differently than a low alert other than the threshold and the direction of the trigger?

**Comment by lfagotbouquet** [comment]
By the way I found this morning how to get the notification logs on my smartphone, so I'm going to do more tests in the following days and check this log.

Also I'm wondering if it could be possible that I'm awaken by the high alert but able to snooze it "unconsciously"...
Because I can fall asleep again very easily for a high alert, but this is not the case for a low alert due to the physical effects of the hypoglycemia...
Even if I don't think it's the correct explanation I'm going to check that.

**Reply by Navid200** [reply]
When an alert goes off, there is a corresponding log added.  So, if you see the log, that tells you that the alert triggered and when.

**Reply by lfagotbouquet** [reply]
Ah I was talking about the android notification log (to check if there was some differences in the metadata for the high and low alerts).
But I just checked the log in the debug mode of xdrip+... and indeed some high alerts are raised during the night with no sound or vibration (one was producing last night an alert event every 1 minutes for almost 1 hour as my glucose level went a bit down after this amount of time, no snoozing event was reported of course).

So at least it's not me that snooze unconsciously the high alerts ^^
I'm going to continue my investigations in the following days.

**Reply by Navid200** [reply]
Can you post a screenshot of the log so that we know exactly what you see.

**Reply by lfagotbouquet** [reply]
Here are some screenshots for an alert last night
![Screenshot_20220520-170752](https://user-images.githubusercontent.com/54840765/169558358-4e53816c-32ed-4963-9f2b-efab57f12ff8.png)
![Screenshot_20220520-170951](https://user-images.githubusercontent.com/54840765/169558377-8e210d60-ec98-417b-b79e-7d04da2030a2.png)
![Screenshot_20220520-170832](https://user-images.githubusercontent.com/54840765/169558382-37b58fe3-c43e-4ec8-9ed6-c37544afcb1e.png)
![Screenshot_20220520-170821](https://user-images.githubusercontent.com/54840765/169558392-69e3f44b-cefe-488c-87ff-519789fd21f2.png)
![Screenshot_20220520-170732](https://user-images.githubusercontent.com/54840765/169558401-943d664a-aa83-4686-99f7-9c5255455f3b.png)

**Comment by Navid200** [comment]
What volume profile are you using?
Do you hear the high alert if it goes off during the day?

**Reply by lfagotbouquet** [reply]
The alert volume profile is set to high in xdrip

I can hear and get vibrations for the high alert during the day, but during the night the high alerts notification are completely silent (no sound and no vibration)
But I can still hear the low level alarms during the night, which is strange as they have the same settings. Maybe they don't have the same priority and I have some kind of do not disturb mode or battery optimization that take that into account ? (I tried to remove completely the do not disturb mode and the battery optimization)

I m waiting for a low level alert to compare the metadata of the high and low level alerts in the android notification log, and I m going to check again my smartphone settings to see if a "smart" mode of any kind is running at night that could impact the notifications

**Reply by Navid200** [reply]
Let's wait to see if @tzachi-dar has any ideas.
If not, would you please open an issue because I have run out of ideas.
@tolot27

**Comment by lfagotbouquet** [comment]
After some new tests during the past week, I found that this issue is not related with the High or Low alert category...
Actually I was using a sound named "Beep-Beep-Beep Alarm" for the Low alerts and "BeepBeep Alarm" for the High ones (to have a longer sound for low ones).
And if I'm using the "Beep-Beep-Beep Alarm" for the High alerts, then my phone is ringing and vibrating during the night for these alerts as well.

So it seems to be related with the alarm sound, maybe at night when my phone is unused for a long time only a long alarm is able to "activate" it again? (I'm not at all an expert on Android so I'm very speculative here...)
And it's surely more an issue with Android or my phone hardware and not xdrip itself.
I also tried to look into the code and even though high and low alerts are handled a bit separately, at the end the notification seems to be sent in a similar way.

It's also not a critical issue for me anymore as I found a workaround for me.
It's still a bit strange that the "BeepBeep alarm" are only working during the day on my phone (and with an old phone with Android 9 it was working fine with this sound during the night).

Thanks for everyone that spent some time on this issue to help me.
